Fixes tests for #60

ggmarshall opened the pull request.

ggmarshall commented:
tests pass with lgdo 1.11.7 but not with pre-release, need to look into why

gipert commented:
I didn't realise you had fixes for the new lgdo here and I merged @schwarzmario's #56... sorry

ggmarshall commented:
oh god, ok i take a look

ggmarshall commented:
is the bound on lgdo set correctly on the last release ?

ssailer commented:
The last 3 commits add support for changes upstream in libfsp (also added to Orca now).
Mainly  some parameters have been upgraded to arrays from scalars, to allow differing settings between channels.
Some have not been activated yet in the user-code facing API. I.e. prescaling ratios can be set for each "GE" channel separately.
Additional observables prepare the use-case, where prescaling would be used in `phy` mode, and tracks the triggered flag per channel. Unlikely to be used soon, but the dataformat supports this usecase now.

In addition, the decoding of FCIO files now supports separating the fsp flags into separate tables depending on triggered or not triggered record (i.e. event vs eventheader record). 
This should allow applying the flags directly to the read out channel tables. 

The fcio dependency version was bumped to reflect these changes.

@ggmarshall Maybe you could test this branch with p14/r003 data?

The following dictionaries are exemplary to decode all data contained into the desired tables. 

```
out_spec_fc = {
  "FCConfigDecoder": {
    "fc_config": {
      "key_list": ["*"],
      "out_stream": "{filekey}:/extra",
    }
  },
  "FSPConfigDecoder": {
    "fsp_config": {
      "key_list": ["*"],
      "out_stream": "{filekey}:/swt",
    }
  },

  "FCEventDecoder": {
    "ch{key:07d}/raw": {
      "key_list": ["*"],
      "out_stream": "{filekey}:/daq",
    },
  },
  "FCEventHeaderDecoder": {
    "evt_hdr": {
      "key_list": ["*"],
      "out_stream": "{filekey}:/swt",
    }
  },

  "FSPEventDecoder": {
    "fsp_evt": {
      "key_list": ["fsp_event_*"],
      "out_stream": "{filekey}:/swt",
    },
    "fsp_evt_hdr": {
      "key_list": ["fsp_eventheader_*"],
      "out_stream": "{filekey}:/swt",
    },
  },

  "FCStatusDecoder": {
    "status/card{key:07d}": {
      "key_list": ["*"],
      "out_stream": "{filekey}:/extra",
    }
  },
  "FSPStatusDecoder": {
    "fsp_status": {
      "key_list": ["*"],
      "out_stream": "{filekey}:/swt",
    }
  },
}

out_spec_orca = {
  "ORFCIOConfigDecoder": {
    "fc_config_{key}": {
      "key_list": ["*"],
      "out_stream": "{filekey}:/extra",
    },
    "{key}": {
      "key_list": ["fsp_config_*"],
      "out_stream": "{filekey}:/swt",
    }
  },
  "ORFCIOEventDecoder": {
    "ch{key:07d}/raw": {
      "key_list": ["*"],
      "out_stream": "{filekey}:/daq",
    },
    "{key}": {
      "key_list": ["fsp_event_*"],
      "out_stream": "{filekey}:/swt",
    },
  },
  "ORFCIOEventHeaderDecoder": {
    "evt_hdr_{key}": {
      "key_list": ["*"],
      "out_stream": "{filekey}:/swt",
    },
    "{key}": {
      "key_list": ["fsp_eventheader_*"],
      "out_stream": "{filekey}:/swt",
    },
  },
  "ORFCIOStatusDecoder": {
    "status/card{key:07d}": {
      "key_list": ["*"],
      "out_stream": "{filekey}:/extra",
    },
    "{key}": {
      "key_list": ["fsp_status_*"],
      "out_stream": "{filekey}:/swt",
    },
  },
}
```

ggmarshall commented:
great thanks simon, will test tomorrow

ggmarshall commented:
Ok I have added skimmed orca files from p14 and p13 to the testdata and added some quick unit tests here based on the old ones. I think in general our unit tests are really lacking and should be as close to a full packet to packet check of orca to lh5 as possible but I think this is ok for now and we can probably go ahead with merging and tagging this now

ggmarshall commented:
numpy 2 .....
